src/lock.js

```diff
@@ -0,0 +1,36 @@
+class Lock {
+  constructor(){
+    this.locked = new Map();
+  }
+
+  acquire(key, ttl){
+    let locked = this.locked.get(key);
```

Comment:
`ttl` param is not used. Is it by design?

Reply:
For a single process, ttl is not necessary. So I didn't add the ttl mechanism. I will comment it.

Reply:
OK